### PR TITLE
Add KCL icon

### PR DESCRIPTION
**Pull request by dennybaa**
image

Add KCL lang icon.

Issue: N/A
GitHub stars: [1.7k](https://github.com/kcl-lang/kcl)

KCL is an open-source, constraint-based record and functional language that enhances the writing of complex configurations, including those for cloud-native scenarios.

### Checklist

- [x] I updated the JSON data in `_data/simple-icons.json`
- [x] I optimized the icon with SVGO or SVGOMG
- [x] The SVG `viewbox` is `0 0 24 24`

### Description

The color used is chosen from one the colors of official logo.

### CODE REVIEW

**Comment by uncenter** [comment]
Unfortunately KCL does not meet our popularity metric requirements (listed at https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#assessing-popularity) under the GitHub stars criteria or the Similarweb ranking (https://www.similarweb.com/website/kcl-lang.io/). Going to mark this as a draft and out of scope for now, sorry! Please feel free to comment on this issue down the road if/when KCL meets our requirements and we can add it.

File: icons/kcl.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" data-name="图层 1"><path d="M18.89 2.134 10.5 5.942a.659.659 0 0 0-.379.51l-.702 5.165a.655.655 0 0 0 .907.69l8.275-3.535a.659.659 0 0 0 .39-.503l.818-5.436a.656.656 0 0 0-.92-.699ZM.025 1.61l2.538 8.859c.062.217.23.385.447.447l5.007 1.45a.653.653 0 0 0 .814-.794L6.546 2.867a.651.651 0 0 0-.441-.46L.846.803a.655.655 0 0 0-.82.807Zm2.52 10.834 2.537 8.859c.062.217.23.385.447.447l5.007 1.45a.653.653 0 0 0 .814-.794L9.065 13.7a.651.651 0 0 0-.441-.46l-5.259-1.605a.655.655 0 0 0-.82.807Zm21.164 3.548-7.644-5.144a.661.661 0 0 0-.63-.056l-4.765 2.115a.654.654 0 0 0-.112 1.134l7.352 5.187a.66.66 0 0 0 .634.068L23.6 17.14a.657.657 0 0 0 .11-1.15z" style="fill:#000;stroke-width:.310609"/></svg>
```
Comment (uncenter) [inline icons/kcl.svg:1]:
```suggestion
<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>KCL</title><path d="M18.89 2.134 10.5 5.942a.659.659 0 0 0-.379.51l-.702 5.165a.655.655 0 0 0 .907.69l8.275-3.535a.659.659 0 0 0 .39-.503l.818-5.436a.656.656 0 0 0-.92-.699ZM.025 1.61l2.538 8.859c.062.217.23.385.447.447l5.007 1.45a.653.653 0 0 0 .814-.794L6.546 2.867a.651.651 0 0 0-.441-.46L.846.803a.655.655 0 0 0-.82.807Zm2.52 10.834 2.537 8.859c.062.217.23.385.447.447l5.007 1.45a.653.653 0 0 0 .814-.794L9.065 13.7a.651.651 0 0 0-.441-.46l-5.259-1.605a.655.655 0 0 0-.82.807Zm21.164 3.548-7.644-5.144a.661.661 0 0 0-.63-.056l-4.765 2.115a.654.654 0 0 0-.112 1.134l7.352 5.187a.66.66 0 0 0 .634.068L23.6 17.14a.657.657 0 0 0 .11-1.15z"/></svg>
```

As per https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#4-annotate-the-icon and https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#5-check-the-icon, the SVG needs these attributes but no others, and needs the `<title>` element.

**Review by uncenter — CHANGES_REQUESTED** [review]
I'll give this a quick review despite the aforementioned.

**Comment by LitoMore** [comment]
This is still far from 5000 stars, so I'm closing it now.

Feel free to come back once it meets our requirements.

**Pull request closed without merging**